modules/fedimint-mint-client/src/lib.rs

```diff
@@ -212,6 +220,31 @@ impl MintClientExt for Client {
             .await
     }
 
+    async fn await_mint_change(
```

Comment:
How come this needs to be a custom function versus just called on the global api?

Reply:
This needs a `await_primary_module_output_finalized(OutPoint)` function since we don't even know if the change will always be generated by the mint module. This didn't appear super crucial at first since we know we will get the money once the tx is accepted, but should be patched in. We'd also need to return a change `Option<OutPoint>` whenever one might be added.

Reply:
Actually to minimize merge conflicts I will do this in a separate PR. Made issue and assigned myself: https://github.com/fedimint/fedimint/issues/2495. Working on this next.

Reply:
We also will probably need a `get_total_funds` implemented on all modules so we can easily return that to the client for all modules.